src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/AbstractControllerTests.kt

```diff
@@ -19,9 +28,34 @@ class AbstractControllerTests : MontaguTests()
     @Test
     fun `can build public URL`()
     {
-        val c = Controller()
+        val c = Controller(ControllerContext("/v6", mock(), mock()))
         assertThat(c.buildPublicUrl("/fragment/")).endsWith(
             "/v6/test/fragment/"
         )
     }
+
+    @Test
+    fun `can get onetime link token`()
+    {
```

Comment:
I moved this test out of the abstract base class `ControllerTests<T>` as it was being run multiple times (once for each implementation of the class)